utils/oracledb.py

```diff
@@ -0,0 +1,88 @@
+import cx_Oracle
+import yaml
+import os
+from pathlib import Path
+home = str(Path.home())
+
+def process_yaml():
+	with open("../config.yaml") as file:
+		return yaml.safe_load(file)
+
+
+class OracleJSONDatabaseConnection:
+    def __init__(self, data=process_yaml()):
+        # wallet location (default is HOME/wallets/wallet_X)
+        os.environ['TNS_ADMIN'] = '{}/{}'.format(home, process_yaml()['WALLET_DIR'])
+        print(os.environ['TNS_ADMIN'])
+        self.pool = cx_Oracle.SessionPool(data['db']['username'], data['db']['password'], data['db']['dsn'],
+            min=1, max=4, increment=1, threaded=True,
+            getmode=cx_Oracle.SPOOL_ATTRVAL_WAIT
+        )
+        print('Connection successful.')
+
+
+
+    def close_pool(self):
+        self.pool.close()
+        print('Connection pool closed.')
+
+
+
+    def insert(self, collection_name, json_object_to_insert):
+        connection = self.pool.acquire()
+        connection.autocommit = True
+        soda = connection.getSodaDatabase()
+        x_collection = soda.createCollection(collection_name)
+
+        try:
+            x_collection.insertOne(json_object_to_insert)
+            print('[DBG] INSERT {} OK'.format(json_object_to_insert))
+        except cx_Oracle.IntegrityError as e:
+            print('[DBG] INSERT {} ERR: {} '.format(json_object_to_insert, e))
+            return 0
```

Comment:
Not trying to be nit-picky, but this should probably return a non-zero error code in this situation?  0 implies no error... something like 1 (or any other non-zero number) would imply something didn't go right.

Reply:
Changed to -1